src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,887 @@
+     * The era and other fields in {@link ChronoField} are defined by the chronology.
+     *
+     * @return the International fixed chronology, not null
+     */
+    @Override
+    public InternationalFixedChronology getChronology() {
+        return InternationalFixedChronology.INSTANCE;
+    }
+
+    /**
+     * Gets the era applicable at this date.
+     * <p/>
+     * The International fixed calendar system only has one era, 'CE',
+     * defined by {@link InternationalFixedEra}.
+     *
+     * @return the era applicable at this date, not null
+     */
+    @Override
+    public InternationalFixedEra getEra() {
+        return InternationalFixedEra.CE;
+    }
+
+    /**
+     * Returns the length of the month represented by this date.
+     * <p/>
+     * This returns the length of the month in days.
+     * Month lengths do not match those of the ISO calendar system.
+     *
+     * @return the length of the month in days: 28
+     */
+    @Override
+    public int lengthOfMonth() {
+        return InternationalFixedChronology.DAYS_IN_MONTH;
+    }
+
+    /**
+     * Returns the length of the year represented by this date.
+     * <p/>
+     * This returns the length of the year in days.
+     * Year lengths match those of the ISO calendar system.
+     *
+     * @return the length of the year in days: 365 or 366
+     */
+    @Override
+    public int lengthOfYear() {
+        return InternationalFixedChronology.DAYS_IN_YEAR + (isLeapYear() ? 1 : 0);
+    }
+
+    @Override
+    public InternationalFixedDate plus(final long amountToAdd, final TemporalUnit unit) {
+        if (amountToAdd == 0) {
+            return this;
+        }
+
+        if (unit instanceof ChronoUnit) {
+            ChronoUnit f = (ChronoUnit) unit;
+
+            switch (f) {
+                case DAYS:
+                    return plusDays(amountToAdd);
+                case MONTHS:
+                    return plusMonths(amountToAdd);
+                case YEARS:
+                    return plusYears(amountToAdd);
+                default:
+                    break;
+            }
+        }
+
+        return (InternationalFixedDate) super.plus(amountToAdd, unit);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    InternationalFixedDate plusYears(final long yearsToAdd) {
+        if (yearsToAdd == 0) {
+            return this;
+        }
+
+        int newYear = (int) Math.addExact(getProlepticYear(), yearsToAdd);
+
+        if (isYearDay()) {
+            return create(newYear, false, true);
+        }
+
+        int day = getDayOfMonth();
+        int newMonth = getCalculatedMonth();
+
+        if (isLeapDay()) {
+            if ((yearsToAdd & 3) == 0) {
+                return create(newYear, true, false);
+            }
+
+            day = 1;
+        }
+
+        return create(newYear, newMonth, day);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public InternationalFixedDate plusMonths(final long months) {
```

Comment:
The reason why I said it would be "surprising" was because:

 1. The documentation declares your flag values (if it doesn't already, it needs to).
 2. The various `ValueRange` instances declare it as a valid value
 3. `get(MONTH)`/`get(DAY_OF_MONTH)` (and the shortcuts) return the flag values.
  
BUT!

> The constructor wouldn't accept the value

Headscratching all around.